packages/etherlime-lib/src/deployer/deployer.ts

```diff
@@ -13,8 +19,13 @@ class Deployer {
 	 * @param {*} provider ethers.provider instance
 	 * @param {*} defaultOverrides [Optional] default deployment overrides
 	 */
-	constructor(signer, provider, defaultOverrides) {
-		this._validateInput(signer, defaultOverrides);
+
+	signer: Wallet;
+	provider: JsonRpcProvider | InfuraProvider | Web3Provider;
```

Comment:
I think, InfuraProvider is a subclass of JSONRPC Provider so you do not need to have Infura here too

Reply:
Not sure but the same may apply for Web3Provider too